react-components/src/components/Architecture/AnchoredDialog.tsx

```diff
@@ -16,21 +16,35 @@ export const AnchoredDialog = (): ReactNode => {
   AnchoredDialogUpdater.setCounterDelegate(setActiveToolUpdater);
 
   const renderTarget = useRenderTarget();
-  if (renderTarget === undefined) {
-    return <></>;
-  }
   const activeTool = renderTarget.commandsController.activeTool;
-  if (activeTool === undefined) {
-    return <></>;
-  }
-
-  const dialogContent = useMemo(() => activeTool.getAnchoredDialogContent(), [activeToolUpdate]);
+  const dialogContent = useMemo(() => activeTool?.getAnchoredDialogContent(), [activeToolUpdate]);
   const isSomeEnabled = dialogContent?.contentCommands.some((command) => command.isEnabled);
 
+  useEffect(() => {
+    if (dialogContent?.customListeners === undefined) return;
+    dialogContent.customListeners.forEach((listener) => {
+      window.addEventListener(listener.eventName, listener.callback, true);
+    });
```

Comment:
Very nitpicky, but this could be a little shorter with `dialogContent.customListeners?.forEach(...` and removing the if-check